platforms/nuttx/src/px4/common/include/px4_platform/micro_hal.h

```diff
@@ -48,6 +48,16 @@ __BEGIN_DECLS
 #define px4_enter_critical_section()       enter_critical_section()
 #define px4_leave_critical_section(flags)  leave_critical_section(flags)
 
+#ifdef CONFIG_SMP
+#define PX4_SCHED_LOCK_TOKEN irqstate_t _sl_flags
```

Comment:
This token looks a bit fuzzy. px4_sched_lock could jus always return flags and unlock take it as argument... Let compiler optimize them away when SMP? Just suggesting, this works of course.

Reply:
Do you mean like this?

```
#define px4_sched_lock()  irqstate_t __flags = px4_enter_critical_section()
#define px4_sched_unlock() leave_critical_section(__flags)

int foo(void)
{
  sched_lock();
  
  do_something_that_needs_sched_lock();
  
  sched_unlock();
  
  ...
  
  sched_lock(); // Multiple definition of __flags
  
  do_something_else_that_needs_sched_lock();
  
  sched_unlock();
  
  return 0;
}
```

How do you guarantee unique name for the flags token? You can't, and will get a multiple definition of X error from the compiler.

Reply:
no, I ment that the px4_sched_lock can just be like:
sched_lock_flags_t flags = px4_sched_lock();
...
px4_sched_unlock(flags);

If you don't use "flags" for anything in SMP shouldn't matter; if px4_sched_lock is an inline function it can just return 0 in SMP mode, and unlock doesn't need to use the argument -> they get optimized away.

Reply:
I removed the whole patch, I'm not convinced sched_lock() was enough in flat mode either -> use critical section there as well.